### ISSUE TITLE
Performance Checker Xcode 14

### ISSUE THREAD

**Issue by daniapopov** [amplitude/Amplitude-iOS#413]
## Expected Behavior 
DispatchSemaphore seems like it isn't working as intended here

## Current Behavior
It seems like there is a queue prioritization bug
Console output is:
```
Performance Checker: Thread running at QOS_CLASS_USER_INITIATED waiting on a lower QoS thread running at QOS_CLASS_UTILITY. Investigate ways to avoid priority inversions
```
<img width="1045" alt="image" src="https://user-images.githubusercontent.com/73848386/193991963-f0728706-cce7-4c7c-ad87-60e4cb1fcf30.png">

## Steps to Reproduce
Run in Xcode 14

## Environment
- SDK Version: v.8.13.0 
- OS Version: MacOS 10.12.6 Monterey

**Comment by bohan-amplitude** [comment]
Hi @daniapopov , Thank you for your feedback. I'll look into this and get back to you.

**Comment by qingzhuozhen** [comment]
Hi @daniapopov, thanks for raising this up. This may be for the recent change in Xcode 14. Found a related [article](https://developer.apple.com/documentation/xcode/diagnosing-performance-issues-early). I tested with Xcode 14.0.1 and v8.14 and wasn't able to reproduce the warning. Would you mind sharing a bit more on the steps to reproduce? Is this still constantly happening for you now?
cc @bgiori @liuyang1520 @yuhao900914

**Comment by liuyang1520** [comment]
Thank you for the feedback! I cannot reproduce in my local with:
- Xcode `Version 14.1 (14B47b)`
- `Thread Performance Checker` enabled
- Amplitude v8.14.0
Close this due to inactivity.

**Comment by daniapopov** [comment]
Still seeing this with the latest Amplitude SDK:
<img width="483" alt="image" src="https://github.com/amplitude/Amplitude-iOS/assets/73848386/7f6c286b-4bff-468e-8b17-7680dee26b4d">
I am running this w/ Performance Checker enabled in the Scheme Editor:
<img width="339" alt="image" src="https://github.com/amplitude/Amplitude-iOS/assets/73848386/7cde3623-95dd-41ef-bdf4-429e95ed55e1">

You don't need to do this. But it might help to have this break point enabled as well:
<img width="656" alt="image" src="https://github.com/amplitude/Amplitude-iOS/assets/73848386/db9fcec7-6ec1-479c-ba03-4643f4eb667a">

**Comment by daniapopov** [comment]
@qingzhuozhen @liuyang1520 I missed that this issue was closed. Can we reopen this?

**Comment by qingzhuozhen** [comment]
cc @justin-fiedler

**Comment by justin-fiedler** [comment]
Hi @daniapopov we are in process of making the new Swift SDK our recommended latest SDK.
* https://www.docs.developers.amplitude.com/data/sdks/ios-swift/
* https://www.docs.developers.amplitude.com/data/sdks/ios-swift/migration/

The older iOS SDK is being transitioned to maintenance mode and will not receive as frequent as updates.

We will try to address this issue but it may take a while. In the meantime we recommend updating to the Swift SDK for the latest features and bug fixes.

Thank you!

**Comment by daniapopov** [comment]
@justin-fiedler 
Excited that you guys are migration to the new SDK, and we are going to upgrade. But in my first quick test it looks like this issue comes from the shared package that both the new and old sdk depend on.
This stack trace clearly shows the issue is inside Analytics Connector.... and the new SDK has the same issue

**Comment by daniapopov** [comment]
With the new SDK it looks like the issue happens during Identity:
<img width="556" alt="image" src="https://github.com/amplitude/Amplitude-iOS/assets/73848386/bcb46244-3f38-4a0b-8d6b-50d30744ccf0">

**Comment by daniapopov** [comment]
Created new issue in Swift SDK:
https://github.com/amplitude/Amplitude-Swift/issues/90
